src/services/chore.service.js

```diff
@@ -0,0 +1,58 @@
+const Chore = require("../models/chore.model");
+const Preference = require("../models/preference.model");
+const Assignment = require("../models/assignment.model");
+
+const RoomService = require("./room.service");
+
+exports.createChoreAndPreferences = async function(body) {
+  let objects = {};
+
+  //Create chore
+  const chore = new Chore({
+    roomId: body.roomId,
+    name: body.name,
+    time: body.time,
+    recurring: body.recurring
+  });
+
+  await chore.save();
+  objects["chore"] = chore;
+
+  //Get users in the room that the chore was created for
+  userIds = await RoomService.getUserIdsByRoomId(chore.roomId);
+
+  objects["preferences"] = [];
+  //create preferences for chore
+  for (var i = 0; i < userIds.length; ++i) {
+    const preference = new Preference({
+      choreId: chore.id,
+      userId: userIds[i]
+    });
+    await preference.save();
+    objects["preferences"].push(preference);
+  }
+
+  return objects;
+};
+
+exports.getChoreIdsByRoomId = async function(roomId) {
+  chores = await Chore.find({ roomId: roomId });
+  choreIds = [];
+  for (var i = 0; i < chores.length; i++) {
+    await choreIds.push(chores[i].get("_id"));
```

Comment:
choreIds.push does not return a promise you can `await`. Instead you need to do:
```
const id = await chores[i].get("_id")
choreIds.push(id)
```

Reply:
Also is it possible to just do ```Document._id```?

Reply:
also I looked at the documentation, you don't need to await for chores[i].get because it doesn't return a promise. So you can just do ```choreIds.push(chores[i].get("_id"))```